drivers/pinmux/pinmux_stm32.c

```diff
@@ -46,15 +46,26 @@ const struct device * const gpio_ports[STM32_PORTS_MAX] = {
 static int stm32_pin_configure(uint32_t pin, uint32_t func, uint32_t altf)
 {
 	const struct device *port_device = gpio_ports[STM32_PORT(pin)];
```

Comment:
this function should actually do some bounds checking, ie

```c
const struct device *port_device;

if (STM32_PORT(pin) >= STM32_PORTS_MAX) {
        return -EINVAL;
}

port_device = gpio_ports[STM32_PORT(pin)];
```

Reply:
Exact, done